app/models/piece.rb

```diff
@@ -4,10 +4,25 @@ class Piece < ApplicationRecord
 
   def move!(x, y)
     return false unless valid_move?(x, y)
-    update_attributes(x_position: new_x, y_position: new_y)
+
+    if castling_kingside(x, y)
+      rook_at(7, y).update_attributes(x_position: 5, y_position: y)
+    end
+    if castling_queenside(x, y)
+      rook_at(0, y).update_attributes(x_position: 3, y_position: y)
+    end
+    update_attributes(x_position: x, y_position: y)
     # Move.create(piece_id: id)
```

Comment:
So, this line should probably be uncommented and implemented, as your tests are technically passing, but since currently no move records are _ever_ created, it's not fully testing castling.